hypervisor/arch/x86/guest/vlapic.c

```diff
@@ -908,26 +846,22 @@ vlapic_process_eoi(struct acrn_vlapic *vlapic)
 	isrptr = &lapic->isr[0];
 	tmrptr = &lapic->tmr[0];
 
-	/* i ranges effectively from 7 to 0 */
-	for (i = 8U; i > 0U; ) {
-		i--;
-		bitpos = (uint32_t)fls32(isrptr[i].v);
-		if (bitpos != INVALID_BIT_INDEX) {
-			if (vlapic->isrvec_stk_top == 0U) {
-				panic("invalid vlapic isrvec_stk_top %u",
-					vlapic->isrvec_stk_top);
-			}
-			isrptr[i].v &= ~(1U << bitpos);
-			vector = (i * 32U) + bitpos;
-			dev_dbg(ACRN_DBG_LAPIC, "EOI vector %u", vector);
-			vlapic_dump_isr(vlapic, "vlapic_process_eoi");
-			vlapic->isrvec_stk_top--;
-			vlapic_update_ppr(vlapic);
-			if ((tmrptr[i].v & (1U << bitpos)) != 0U) {
-				/* hook to vIOAPIC */
-				vioapic_process_eoi(vlapic->vm, vector);
-			}
-			break;
+	/* vlapic->isrvec_stk[0] always is 0 */
+	if (vlapic->isrvec_stk_top >= 1U) {
+		vector = vlapic->isrvec_stk[vlapic->isrvec_stk_top];
+		i = (vector >> 5U);
+		bitpos = (vector & 0x1fU);
+		isrptr[i].v &= ~(1U << bitpos);
+
+		dev_dbg(ACRN_DBG_LAPIC, "EOI vector %u", vector);
+		vlapic_dump_isr(vlapic, "vlapic_process_eoi");
+
+		vlapic->isrvec_stk_top--;
```

Comment:
With bitpos = (uint32_t)fls32(isrptr[i].v);, We already have bitpos, which is the highest in service vector too. 
I thought Anthony want completely remove isrvec_stk_top.
Not?

Reply:
will wait and merge these 2 patches into one. It is simple one.

Reply:
@dongyaozu  Merged. Could you help to have a review again ?